fix: nushell supports darwin config file locations

davidarmstronglewis opened the pull request:
### Description

Updates the nushell program to support the correct config file locations on Darwin systems.

### Checklist

<!--

Please go through the following checklist before opening a non-WIP
pull-request.

Also make sure to read the guidelines found at

  https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-guidelines

-->

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).
  - No existing tests exist, none added

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

davidarmstronglewis commented:
@rycee I'm not sure where I should be asking for a review. Is there a forum for these kinds of PR review requests I should be using?

rycee commented on modules/programs/nushell.nix:
```diff
@@ -6,6 +6,26 @@ let
+  configDir = if pkgs.stdenv.isDarwin then
+    "Library/Application Support/nushell"
+  else
+    "nushell";
```
Can use `home.file` for both Darwin and GNU/Linux by giving the full path here:

```suggestion
    "${config.xdg.configHome}/nushell";
```

rycee commented on modules/programs/nushell.nix:
```diff
@@ -111,19 +131,7 @@ in {
   config = mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
-    xdg.configFile = mkMerge [
```
Change this to `home.file` and …

rycee commented on modules/programs/nushell.nix:
```diff
@@ -111,19 +131,7 @@ in {
 
-    xdg.configFile = mkMerge [
-      (mkIf (cfg.configFile != null || cfg.extraConfig != "") {
-        "nushell/config.nu".text = mkMerge [
```
… this to `"${configDir}/config.nu"` and …

rycee commented on modules/programs/nushell.nix:
```diff
@@ -111,19 +131,7 @@ in {
-        ];
-      })
-      (mkIf (cfg.envFile != null || cfg.extraEnv != "") {
-        "nushell/env.nu".text = mkMerge [
```
… this to `"${configDir}/env.nu".text = mkMerge [`.

rycee commented:
@davidarmstronglewis Thanks for the contribution! I added a few comments.

Sorry, there is no particular place to ask for reviews other than possibly the IRC channel.

In general the PR notification will end up in the maintainer mail boxes (where mine currently contains 2860 unread new mails). So it is mainly a matter of being patient :-)

rycee commented:
Oh, and you'll need to update the [test case](https://github.com/nix-community/home-manager/blob/37713c6b04b963d41664e03576f73a18c9b0d224/tests/modules/programs/nushell/example-settings.nix) as well. See https://github.com/nix-community/home-manager/blob/37713c6b04b963d41664e03576f73a18c9b0d224/tests/modules/programs/bottom/example-settings.nix#L21-L30 for an example.

davidarmstronglewis commented:
@rycee Thank you so much for the PR feedback! I think I've addressed the points you've called out - let me know if I missed anything or if something else jumps out.

Appreciate the pointer to the IRC channel. I probably won't be a regular as I'm overloaded already with GitHub+Slack+Discord+Mastodon already when it comes to tech communities & async communication, but will definitely drop in when I have something to contribute ❤️

rycee commented:
Thanks! Merged to master now.
